### PR TITLE
Integration test for webapi

### PR DESCRIPTION
**Pull request by johevemi**
## Description

The PR will add the following:

- IntegrationTests and IntergrationTests.Core projects
- IntegrationTests.Core will implement TestCommon and create the ability to make fixture for IntegrationTests
- IntegrationTests has a simple test of the StartProcessController, which was created in #13 
- Include IntegrationTest in CI

I have added the relevant secret to run integration test to the github repo.

Secrets added:
- AZURE_TENANT_ID
- AZURE_SUBSCRIPTION_ID
- AZURE_SPN_ID
- AZURE_SPN_SECRET
- AZURE_KEYVAULT_URL
- AZURE_SECRETS_KEYVAULT_URL

In my upcoming work, I will implement JWT token authentication
## References

* [#765](https://app.zenhub.com/workspaces/joules-5fd22a1a59e4740018f5ab5a/issues/energinet-datahub/geh-aggregations/765)

### CODE REVIEW

File: source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    public Task DisposeAsync()
+    {
+        _client.Dispose();
+        return Task.CompletedTask;
+    }
+    
+    [Fact]
+    public async Task StartProcessAsync()
```
Comment (lasrinnil) [inline source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs:57]: The test name does not describe what it tests.

consider something like this formula used in the charges domain:

`GetAsync_WhenMeteringPointIdHasChargeLinks_ReturnsOkAndCorrectContentType`

If you need help finding a suitable name, let me know :)

File: source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        // Act
+        var response = await _client.PostAsync(BaseUrl, new StringContent("test"), CancellationToken.None);
+
+        // Assert
+        response.StatusCode.Should().Be(HttpStatusCode.OK);
```
Comment (lasrinnil) [inline source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs:63]: I would much prefer we from the beginning started using Fluent assertions, for more information please see: [https://fluentassertions.com/](url)

**Review by lasrinnil — COMMENTED** [review]
Great work! Now that we have a integration test, we should build the solution and run the test of our CI pipeline 👍 🚀

File: source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs (hunk 1 above)
Comment (johevemi) [reply source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs]: Name changed to StartProcessAsync_ReturnsOk and suppressed [VSTHRD200](https://github.com/microsoft/vs-threading/blob/main/doc/analyzers/VSTHRD200.md) warning

File: source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs (hunk 2 above)
Comment (johevemi) [reply source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs:65]: > Great work! Now that we have a integration test, we should build the solution and run the test of our CI pipeline 👍 🚀

Yes I will implement this in CI

**Comment by johevemi** [comment]
> Great work! Now that we have a integration test, we should build the solution and run the test of our CI pipeline 👍 🚀

Yes, I will implement this in CI

**Review by BjarkeMeier — COMMENTED** [review]
Only minor comment(s) aside from I'd also like to see the test execute in CI and blocking PR gate on failure.

File: source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs (hunk 1 above)
Comment (johevemi) [reply source/dotnet/IntegrationTests/IntegrationTests/WebApi/V1/StartProcessControllerTests.cs]: Added .editorconfig
image
This is the way

File: source/dotnet/IntegrationTests.Core/Fixtures/WebApi/WholesaleWebApiFixture.cs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Energinet.DataHub.Wholesale.IntegrationTests.Core.Fixtures.WebApi
+{
+    public class WholesaleWebApiFixture : WebApiFixture
```
[the remark this answers is not in the document]
Comment (johevemi) [reply source/dotnet/IntegrationTests.Core/Fixtures/WebApi/WholesaleWebApiFixture.cs:19]: WholesaleWebApiFixture is there to implement WebApiFixture (WebApiFixture will in time be part of TestCommon) and make it possible to override behavior. Right now it is just a default implementation.

**Review by BjarkeMeier — APPROVED** [review]
Please let create-prerelease step in CI depend on the test execution.

**Pull request merged**